docs/synapse/userguides/ug015_storm_ref_pivot.rst

```diff
@@ -11,85 +15,339 @@ Where specific query examples are given, they are commonly provided in pairs usi
+    pivot( inet:fqdn, inet:dns:a:fqdn ) pivot( inet:dns:a:ipv4, inet:ipv4 )
+      
+    pivot( inet:dns:a:fqdn ) pivot( :ipv4, inet:ipv4 )
+    
+    inet:fqdn -> inet:dns:a:fqdn inet:dns:a:ipv4 
+      -> inet:ipv4
+    
+    -> inet:dns:a:fqdn :ipv4 -> inet:ipv4
+
+* Pivot from a set of domains in the working set to the set of subdomains for those domains:
+  ::
+    pivot( inet:fqdn, inet:fqdn:domain )
+    
+    pivot( inet:fqdn:domain )
+    
+    inet:fqdn -> inet:fqdn:domain
+    
+    -> inet:fqdn:domain
+
+* Pivot from a set of email addresses in the working set to the set of domains registered to those email addresses.
+  ::
+    pivot( inet:email, inet:whois:regmail:email )
+      pivot( inet:whois:regmail:fqdn, inet:fqdn )
+    
+    pivot( inet:whois:regmail:email ) pivot( :fqdn, inet:fqdn )
+    
+    inet:email -> inet:whois:regmail:email 
+      inet:whois:regmail:fqdn -> inet:fqdn
+    
+    -> inet:whois:regmail:email :fqdn -> inet:fqdn
+
+* Pivot from a set of email addresses in the working set to the set of whois records associated with those email addresses.
+  ::
+    pivot( inet:email, inet:whois:contatct:email )
+      pivot( inet:whois:contact:rec, inet:whois:rec )
+    
+    pivot( inet:whois:contact:email ) pivot( :rec, inet:whois:rec )
+    
+    inet:email -> inet:whois:contact:email inet:whois:contact:rec
+      -> inet:whois:rec
+    
+    -> inet:whois:contact:email inet:whois:contact:rec -> inet:whois:rec
+
+**Usage notes:**
+
+* If the source property for the pivot is the primary property of the working set of nodes, the *<srcprop>* can be omitted from Operator syntax. The *<srcprop>* can also be omitted from Macro syntax, unless a limit parameter ( ``^`` ) is specified.
+* If the source property for the pivot is a secondary property of the working set of nodes, relative property syntax can be used to specify *<srcprop>* as the source properties are, by definition, properties from the working set of nodes.
+* The ``limit=`` parameter can be provided as input to the ``pivot()`` operator itself; alternately the ``limit()`` operator_ can be used after the ``pivot()`` operator to specify a limit on the number of nodes returned.
 
 join()
 ------
 Todo
 
 refs()
 ------
-Todo
+Returns the set of nodes that "reference" or are "referenced by" the working set of nodes.
+
+Optional parameters:
+
+* **in:** return all nodes that have a secondary property *<type> (<ptype>) = <valu>* that is the same as (**references**) any primary *<prop> = <valu>* in the working set of nodes.
+* **out:** return all the nodes whose primary *<prop> = <valu>* is the same as (is **referenced by**) any secondary property *<type> (<ptype>) = <valu>* in the working set of nodes.
+* If no parameters are specified, ``refs()`` will return the combined results of both ``refs(in)`` and ``refs(out)`` (e.g., execute all pivots to / from the working set of nodes).
+
+**Operator syntax:**
+
+.. parsed-literal::
+  **refs(** [ **in** | **out** ] **)**
```

Comment:
`limit` also works on refs

Reply:
Updated refs() documentation / usage statement / usage notes to reflect "current state", including non-consumption of nodes and that effect limit= vs. limit() results.